triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -1118,7 +1142,27 @@ object Runner {
     }
   }
 
+  private def mapSize(smap: SValue): Int = {
+    smap.expect("SMap", { case SMap(_, values) => values.size }).getOrElse(0)
+  }
+
+  // The following method should be kept in sync with ACS in Internal.daml
+  private def numberOfActiveContracts(svalue: SValue): Int = {
+    // svalue: TriggerState s
+    val result = for {
+      acs <- svalue.expect("SRecord", { case SRecord(_, _, values) => values.get(0) })
+      activeContracts <- acs.expect("SRecord", { case SRecord(_, _, values) => values.get(0) })
+      size <- activeContracts.expect(
+        "SMap",
+        { case SMap(_, values) => values.values.map(mapSize).sum },
+      )
+    } yield size
+
+    result.getOrElse(0)
```

Comment:
again I think it will be better to fails instead of return `0`

Reply:
Doing this actually teased out an extra case (i.e. low level trigger state) that I hadn't thought about. So thanks for this.